### PR TITLE
Add test for LobbyGameTableModel, add corner-case fixes and some simplification/reduction

### PR DESCRIPTION
**Pull request by DanVanAtta**
A test for LobbyGameTableModel showed it doesn't handle messages that have NULL id, are out of order or duplicated. To fix that the add and update both become "add or update" operations. The second commit of this PR is to then remove the double implementation and have "gameAdded" call "gameUpdated"

This PR is based on other PRs:  #110

[Review on Reviewable](https://reviewable.io/reviews/triplea-game/triplea/111)

### CODE REVIEW

File: src/games/strategy/engine/lobby/client/ui/LobbyGameTableModel.java [hunk 1]
```diff
@@ -54,47 +60,58 @@ public void gameRemoved(final GUID gameId) {
...
   private void removeGame(final GUID gameId) {
     SwingUtilities.invokeLater(new Runnable() {
       @Override
       public void run() {
-        final int index = m_gameIDs.indexOf(gameId);
-        m_gameIDs.remove(index);
-        m_games.remove(index);
-        fireTableRowsDeleted(index, index);
+        if (gameId == null) {
+          return;
```
Comment (ron-murhammer) [inline src/games/strategy/engine/lobby/client/ui/LobbyGameTableModel.java]: For the new null checks on bad input, is it better to return silently or throw an exception?

File: src/games/strategy/engine/lobby/server/LobbyGameController.java [hunk 2]
```diff
@@ -66,10 +66,10 @@ public void postGame(final GUID gameID, final GameDescription description) {
     synchronized (m_mutex) {
       m_allGames.put(gameID, description);
     }
-    m_broadcaster.gameAdded(gameID, description);
+    m_broadcaster.gameUpdated(gameID, description);
```
Comment (ron-murhammer) [inline src/games/strategy/engine/lobby/server/LobbyGameController.java]: This change is the only one I'm not so sure about. As this is starting to potentially change public calls around lobby functionality. I believe since the method already existed and properly handles the "game added" situation that it shouldn't cause any issues but will need @veqryn to review this.

**Comment by ron-murhammer** [comment]
@DanVanAtta Some good refactoring and unit tests. I looked it over and looks pretty good and I just had the 2 comments above. It appears this needs re-based though.

**Comment by veqryn** [comment]
Some problems, see my comments here: https://github.com/triplea-game/triplea/pull/110
I will look into this PR more after it gets rebased.

**Comment by veqryn** [comment]
Why closed?
Were you going to update to List+Map like we talked about in https://github.com/triplea-game/triplea/pull/110 ?

**Comment by DanVanAtta** [comment]
Branches that were not based on #151, I had to recreate. Deletion of this branch wasn't intentional, casualty of fixing all the branches in mass.

> Were you going to update to List+Map like we talked about in #110 ?

I would like to, but given other priorities, and also that the update is not urgent, I'm not sure when I will be able to.
